### PR TITLE
Complete Reading Logs Utils + Restructuring

### LINKED ISSUE #77: Complete reading log utils

### PR DESCRIPTION
**Pull request by Opeyem1a**
+ complete reading log duration util functions
+ they use real data now lads
+ all reading logs related scripts should go in `scripts/reading_logs` I think

Closes #77

### CODE REVIEW

File: util/reading_logs.py [hunk 1]
```diff
@@ -78,20 +96,39 @@ def get_num_pages_in_module(self, module_num: int) -> int:
...
 
+    def page_reading_duration(self, module_num: int, page_num: int, data448_id: int = None) -> float:
+        """Returns the page reading duration in minutes. Average of all students unless given a data_448 id."""
+        self.reading_duration_dict, _ = self.get_parsed_reading_log_data()
+
+        # Retrieve the correct DataFrame for the requested page.
+        reading_duration_df = self.reading_duration_dict[f'{module_num}-{page_num}']
+
+        if data448_id:
+            start_time = reading_duration_df['start_time'][f'{data448_id}']
```
Comment (carson-ricca) [inline util/reading_logs.py:107]: The data frame ids could be constant since they are being used more than once?

**Review by carson-ricca — CHANGES_REQUESTED** [review]
Looks good overall, how is the cleaned reading log data being used? Is it being uploaded manually into the data folder when you want to run stuff related to reading logs?

File: util/reading_logs.py (hunk 1 above)
Comment (Opeyem1a) [reply util/reading_logs.py:107]: Like the `'start_time'` should be a constant in this function?

**Comment by Opeyem1a** [comment]
The cleaned reading logs get uploaded to `data/api/canvas/reading_logs` and then `parse_reading_logs.py` will create the dictionaries of dataframes and save the binary for them in `.pkl` files stored in `data/cache`. Then `ReadingLogsData` just reads those `.pkl` files.

File: util/reading_logs.py (hunk 1 above)
Comment (carson-ricca) [reply util/reading_logs.py:107]: Yeah, that's what I was thinking, not necessary but just wanted your thoughts on it?

**Comment by carson-ricca** [comment]
Above comment and conflicts now

**Review by carson-ricca — CHANGES_REQUESTED** [review]
Above comment and conflicts now

**Review by carson-ricca — APPROVED** [review]
LGTM now.

**Review by MtheDV — APPROVED** [review]

**Pull request merged**